opencti-platform/opencti-graphql/src/graphql/authDirective.js

```diff
@@ -6,6 +6,45 @@ import { defaultFieldResolver } from 'graphql/index.js';
 import { AuthRequired, ForbiddenAccess, OtpRequired, OtpRequiredActivation } from '../config/errors';
 import { OPENCTI_ADMIN_UUID } from '../schema/general';
 import { BYPASS, VIRTUAL_ORGANIZATION_ADMIN, SETTINGS_SET_ACCESSES } from '../utils/access';
+import { logApp } from '../config/conf';
+
+/**
+ * Retrieves the user's capabilities for an optionally specified entity type.
+ * @param {*} user The user
+ * @param {string?} entityType The entity type to look for in the user's role's overrides.
+ * @returns {Record<string | number | symbol, any>} The user's capabilities
+ */
+const getUserCapabilities = (user, entityType = null) => {
+  let userCapabilities = map((capability) => capability.name, user?.capabilities);
+
+  // Handle bypass
+  if (userCapabilities.includes(BYPASS) || user?.id === OPENCTI_ADMIN_UUID) {
+    return userCapabilities;
+  }
+
+  try {
+    if (entityType) {
+      // Find the overrides of the user's role
+      const overrides = user?.roles?.flatMap((role) => role?.capabilities_overrides);
+
+      // Filter the overrides to be of the specified entity type
+      const filteredOverrides = overrides
+        ?.filter((capability) => capability?.entity.replace(/-/g, '') === entityType);
+
+      // Map the override capabilities to just their names
+      userCapabilities = map(
+        (capability) => capability.name,
+        filteredOverrides?.[0]?.capabilities
```

Comment:
A bit confused about the [0] here

Reply:
filteredOverrides is the overrides property, just filtered down to the specified entity type. The overrides property could contain the overridden capabilities for several entities, but the filter should bring it down to just one. The [0] here is just accessing that filtered entity's overrides.